frame/bags-list/src/migrations.rs

```diff
@@ -17,33 +17,92 @@
 
 //! The migrations of this pallet.
 
-use frame_support::traits::OnRuntimeUpgrade;
+use codec::{Decode, Encode};
+use core::marker::PhantomData;
+use frame_election_provider_support::ScoreProvider;
+use frame_support::{storage::migration, traits::OnRuntimeUpgrade};
+use sp_runtime::traits::Zero;
+
+#[cfg(feature = "try-runtime")]
+use frame_support::ensure;
 
 /// A struct that does not migration, but only checks that the counter prefix exists and is correct.
-pub struct CheckCounterPrefix<T: crate::Config>(sp_std::marker::PhantomData<T>);
-impl<T: crate::Config> OnRuntimeUpgrade for CheckCounterPrefix<T> {
+pub struct CheckCounterPrefix<T: crate::Config<I>, I: 'static>(sp_std::marker::PhantomData<(T, I)>);
+impl<T: crate::Config<I>, I: 'static> OnRuntimeUpgrade for CheckCounterPrefix<T, I> {
 	fn on_runtime_upgrade() -> frame_support::weights::Weight {
-		0
+		frame_support::weights::Weight::zero()
 	}
 
 	#[cfg(feature = "try-runtime")]
 	fn pre_upgrade() -> Result<(), &'static str> {
-		use frame_support::ensure;
 		// The old explicit storage item.
 		frame_support::generate_storage_alias!(BagsList, CounterForListNodes => Value<u32>);
 
 		// ensure that a value exists in the counter struct.
 		ensure!(
-			crate::ListNodes::<T>::count() == CounterForListNodes::get().unwrap(),
+			crate::ListNodes::<T, I>::count() == CounterForListNodes::get().unwrap(),
 			"wrong list node counter"
 		);
 
 		crate::log!(
 			info,
 			"checked bags-list prefix to be correct and have {} nodes",
-			crate::ListNodes::<T>::count()
+			crate::ListNodes::<T, I>::count()
 		);
 
 		Ok(())
 	}
 }
+
+#[derive(Encode, Decode)]
+struct PreScoreNode<T: crate::Config<I>, I: 'static = ()> {
+	pub id: T::AccountId,
+	pub prev: Option<T::AccountId>,
+	pub next: Option<T::AccountId>,
+	pub bag_upper: T::Score,
+	#[codec(skip)]
+	pub _phantom: PhantomData<I>,
+}
+
+/// A struct that migrates all bags lists to contain a score value.
+pub struct AddScore<T: crate::Config<I>, I: 'static>(sp_std::marker::PhantomData<(T, I)>);
+impl<T: crate::Config<I>, I: 'static> OnRuntimeUpgrade for AddScore<T, I> {
+	#[cfg(feature = "try-runtime")]
+	fn pre_upgrade() -> Result<(), &'static str> {
+		// Nothing that can really be checked here...
```

Comment:
You could check the count of nodes staying constant, but that requires writing to storage in per_migrate which is a bit shitty.. at least you can log it?